sagemaker/launch/launch_utils.py

```diff
@@ -0,0 +1,240 @@
+    output_path: str
+        The folder where the output tar archive will be saved. If not provided, will raise an
+        error.
+    output_tarfile_name: str
+        The name of the tar archive. Default is `model`.
+
+    """
+    # check if files exist
+    assert os.path.exists(path_to_model), f'The model file, {path_to_model}, does not exist.'
+    assert os.path.isfile(path_to_model), f'The model file, {path_to_model}, should be a file \
+                                            path, but got a folder.'
+    assert os.path.exists(path_to_yaml), f'The model configuration YAML file, {path_to_yaml}, \
+                                           does not exist.'
+    assert os.path.isfile(path_to_yaml), f'The model configuration YAML file, {path_to_yaml}, \
+                                           should be a file path, but got a folder.'
+    assert os.path.exists(path_to_json), f'The graph metadata JSON file, {path_to_json}, does \
+                                           not exist.'
+    assert os.path.isfile(path_to_json), f'The graph metadata JSON file, {path_to_json}, should \
+                                           be a file path, but got a folder.'
+    # TODO(JianZhang): If entry point file requires modules out of GraphStorm, we need to check
+    #                  and support a list of strings (multiple files) and a folder (multiple files
+    #                  in a folder).
+    assert os.path.exists(path_to_entry), f'The SageMaker entry point file, {path_to_entry}, does \
+                                           not exist.'
+    assert os.path.isfile(path_to_entry), f'The SageMaker entry point file, {path_to_entry}, \
+                                           should be a file path, but got a folder.'
+
+    # create the output folder if not exist.
+    if os.path.exists(output_path):
+        assert os.path.isdir(output_path), 'Output path should be a folder name, but got a ' + \
+                                           f'file {output_path}.'
+    else:
+        os.makedirs(output_path, exist_ok=True)
+    os.makedirs(os.path.join(output_path, 'code'), exist_ok=True)
+
+    # copy the artifacts files to output_path
+    shutil.copy(path_to_entry, os.path.join(output_path, 'code', os.path.basename(path_to_entry)))
+    if output_path != os.path.dirname(path_to_model):
+        shutil.copy(path_to_model, os.path.join(output_path, os.path.basename(path_to_model)))
+    if output_path != os.path.dirname(path_to_yaml):
+        shutil.copy(path_to_yaml, os.path.join(output_path, os.path.basename(path_to_yaml)))
+    if output_path != os.path.dirname(path_to_json):
+        shutil.copy(path_to_json, os.path.join(output_path, os.path.basename(path_to_json)))
+
+    output_file = os.path.join(output_path, output_tarfile_name + '.tar.gz')
+    with tarfile.open(output_file, 'w:gz') as tar:
+        tar.add(output_path, arcname='')
+
+    return output_file
+
+
+def parse_s3_url(s3_url):
```

Comment:
Let's use URI as terminology here and not URL

Reply:
changed all to uri